### PR TITLE
raftstore: fix missing workers' stop on shutdown

### LINKED ISSUE #11159: Support adjusting the pool size of raftstore poller threads dynamically
### Describe the feature you'd like:
A dynamic config modification is performed from TiKV-Control command line ([documentation](https://docs.pingcap.com/tidb/dev/tikv-control#modify-the-tikv-configuration-dynamically)). So expect to adjust the thread pool size in this way:
```sql
set config tikv `raftstore.apply-pool-size`=3
set config tikv `raftstore.store-pool-size`=3
```

### Technical proposal
- Add a `PoolStateBuilder` to `BatchSystem` to create a `PoolState` object to save what is needed to build a `Poller`；
- Define a `RefreshConfigRunner` to handle dynamic configuration updates;
- Define a `PoolController` to save the `PoolState` object and handle dynamically adjusting the pool size of the `Poller`;

### PR DESCRIPTION
**Pull request by ethercflow**
ref #11159

Signed-off-by: Wenbo Zhang <ethercflow@gmail.com>

### What is changed and how it works?

Close #11159  

What's Changed:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

- None

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by NingLin-P — APPROVED** [review]

**Comment by ethercflow** [comment]
cc @tabokie

**Review by tabokie — APPROVED** [review]

File: components/test_raftstore/src/common-test.toml [hunk 1]
```diff
@@ -116,8 +116,3 @@ wake-up-delay-duration = "100ms"
 enable = false
 advance-ts-interval = "10ms"
 scan-lock-pool-size = 1
-[raftstore.store-batch-system]
```
Comment (5kbpers) [inline components/test_raftstore/src/common-test.toml:119]: Why remove these?

Comment (ethercflow) [reply components/test_raftstore/src/common-test.toml:119]: @5kbpers This is an early wrong serialization, it should be kept as `[raftstore] store-pool-size`

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**